src/SonarScanner.MSBuild.PreProcessor/ArgumentProcessor.cs

```diff
@@ -144,12 +144,9 @@ private static bool AreParsedArgumentsValid(ProcessedArgs args, ILogger logger)
                 areValid = false;
             }
 
-            if (!args.IsOrganizationValid)
-            {
-                areValid = false;
-            }
-
-            if (args.SonarServer is null)
+            if (!args.IsOrganizationValid
+                || args.SonarServer is null
+                || !args.IsJavaConfigurationValid)
```

Comment:
We should unify `args.IsOrganizationValid`, `args.SonarServer is null` and `!args.IsJavaConfigurationValid` into a single flag `args.IsValid` And move the logic of `AreParsedArgumentsValid` to ProcessedArgs.cs

Reply:
Yeah I agree that would be an improvement 👍

Reply:
Done in #1984